### ISSUE TITLE
URL / project is not working

### ISSUE THREAD

**Issue by melroy89** [Palatis/Arduino-Lufa#26]
I entered the following url `https://github.com/Palatis/Arduino-Lufa/raw/master/package_arduino-lufa_index.json` as the additional board within the preferences. 

Your URL within the [package_arduino-lufa_index.json](https://github.com/Palatis/Arduino-Lufa/blob/master/package_arduino-lufa_index.json) file is not working. You don't have a version tagged under this Git repo. 

Try for example this link yourself: https://github.com/Palatis/Arduino-Lufa/archive/refs/tags/1.0.zip (it's not found!)

This will most likely gives a lot of problems, since I think the Arduino IDE 2 won't download or install this project. For example I'm also missing the examples (only see built-in examples):

![image](https://github.com/Palatis/Arduino-Lufa/assets/628926/acf66d99-fffc-4b09-87f2-eb00ace6667e)

**Comment by iFreilicht** [comment]
Ah yes, I forgot about this after #23 was merged. I just published a release and the URL now exists. Can you try again?

**Comment by melroy89** [comment]
Nope. It still doesn't work (yes I also restart the IDE everytime).. I also tried: `https://raw.githubusercontent.com/Palatis/Arduino-Lufa/master/package_arduino-lufa_index.json` instead, without any luck.

Checksum is **not** correct, could that cause issues with Arduino IDE? Checksum is now: `61184a377d979cd7aa0b25df92fb828cf173a3e84e388488eaffd9bed2051584`. I expect then: `SHA-256:61184a377d979cd7aa0b25df92fb828cf173a3e84e388488eaffd9bed2051584` as value? Somehow the size does still match..

**Comment by melroy89** [comment]
@iFreilicht anything?

**Comment by iFreilicht** [comment]
Sorry I'm a bit busy at the moment. Maybe @CrazyRedMachine can help out? He made the original PR to add support for the board manager in #23

**Comment by CrazyRedMachine** [comment]
hi, yes as explained in #23 once the release is published you need to update the checksum in the json, i included a link to the relevant lines to be modified

**Comment by CrazyRedMachine** [comment]
@melroy89 you can also just use my fork and its url which has been working for 2 years already https://github.com/CrazyRedMachine/Arduino-Lufa/

i have to say i'm pretty disappointed about the way this merge is being handled... between the delays and the fact nobody in charge seems to read or care about the instructions

**Comment by iFreilicht** [comment]
Ok thank you, I updated the checksum to the value @melroy89 provided. Could you try again?

Sorry about how I've handled this. I took over the maintainer role because I was actively using and improving this library and Victor had no time to take care of the issues and PRs, and so I think it's on me to deal with stuff like this. As I don't do any embedded work right now, this project has low priority for me, but as you basically did all the work and there was very little to do for me, I should've reacted quickly.

**Comment by melroyvandenberg** [comment]
Sorry to hear about all the drama

OK, I will try this repo again soon on my other PC (this is my work account as you may notice as well ahha). 

I also saw that there are libraries within Arduino IDE, allowing to search on libraries and click install. This repo is using a library: https://github.com/NicoHood/HID. Maybe something worth to consider? Just an idea.

**Comment by CrazyRedMachine** [comment]
yea don't worry, looks like it will finally be working now, and the fact it's been used for years prior means we should get little to no issues.

@melroy89 i'm not sure I understand the question ? lufa-arduino cannot be used as a library as it requires to change board definitions to change the usb layer in a way that is conflicting with pluggableusb (the usual arduino usb stack).

**Comment by melroy89** [comment]
Ah I see. Yes in that case ignore my library proposal.

**Comment by melroy89** [comment]
Small question upfront: Am I able to flash the original AVR board (bootloader etc.) without any issues?


I can find and install:

![image](https://github.com/Palatis/Arduino-Lufa/assets/628926/573ad04a-1b49-409d-8e01-6dde48d78696)

Then I install it:

![image](https://github.com/Palatis/Arduino-Lufa/assets/628926/13a52c4c-2f3e-448c-961e-0b61d6210e50)

Which seems to install. But no examples in the menu for LUFA?

![image](https://github.com/Palatis/Arduino-Lufa/assets/628926/71d37ab8-dccc-4803-be22-875d4e581c9b)

**Comment by CrazyRedMachine** [comment]
i think it's normal as examples come from libraries, not boards

as said before, having lufa as library wouldn't work, and having it as a board actually suffices so having an additional library just for the examples would make it confusing?

although it might actually work if we just call the library "lufa examples" and keep mentioning on each example header that you need to download the lufa boards for them to work.. I'll look into it when I have time

in the mean time I suggest taking examples manually from the repo (download zip, extract somewhere then open the .ino of your choice... i also have several repositories making use of lufa, most notable being lufahybridfightstick if you happen to need a gamepad example)

**Comment by melroy89** [comment]
Thanks for your answer, make sense indeed. And about my initial question? I suspect no issues when I need to re-flash again my current old project using Arduino SDK right? So I can safely go back and forward between the two boards?

**Comment by CrazyRedMachine** [comment]
looks like there's only one example which is the dualserial so maybe not worth the trouble.. or i'd need to create more examples from other projects..

https://github.com/Palatis/Arduino-Lufa/blob/master/libraries/Arduino-LUFA/examples/LUFA_DualVirtualSerial/LUFA_DualVirtualSerial.ino

About your first question, yes, you can always flash non-LUFA firmware by selecting the correct board type before flashing (and adding arduino-lufa as custom board was done mainly because i needed to switch efficiently between lufa and non-lufa code)

only thing to note is that lufa won't expose the arduino serial interface (which allows the automatic reset for code upload), therefore you'll have to use the reset button manually during upload whenever you want to flash a new firmware and your board is currently running lufa

**Comment by melroy89** [comment]
> only thing to note is that lufa won't expose the arduino serial interface (which allows the automatic reset for code upload), therefore you'll have to use the reset button manually during upload whenever you want to flash a new firmware and your board is currently running lufa

Ow that is a problem.. Thanks for pointing this out in advance! Since my arduino micro chip is inside.. well.. 😖  See photo:
![usb](https://github.com/Palatis/Arduino-Lufa/assets/628926/a9d202b4-11bb-4fb8-ae6a-3044d68065bd)

**Comment by CrazyRedMachine** [comment]
indeed that makes it annoying... maybe autoreset can work if you take care of adding a cdc interface as first interface in your project and copy the code which triggers a reset to bootloader (should be somewhere in pluggableusb files maybe)

also depending on the chip, i think there's another way to go into bootloader mode which is unplug/replug quickly 3 times on atmega32u4..

**Comment by melroy89** [comment]
Yea, so that issue is worth another issue ticket...

I will close this ticket, since I think the repo is now fixed.
